module/proto/peggy/v1/claims.proto

```diff
@@ -0,0 +1,66 @@
+syntax = "proto3";
+package peggy.v1;
+
+import "gogoproto/gogo.proto";
+
+option go_package = "github.com/althea-net/peggy/module/x/peggy/types";
+
+enum ClaimType {
+  UNKNOWN             = 0;
+  DEPOSIT             = 1;
+  WITHDRAW            = 2;
+  ERC20_DEPLOYED      = 3;
+  LOGIC_CALL_EXECUTED = 4;
+}
+
+// EthereumBridgeDepositClaim
+// When more than 66% of the active validator set has
+// claimed to have seen the deposit enter the ethereum blockchain coins are
+// issued to the Cosmos address in question
+// -------------
+message DepositClaim {
+  uint64 event_nonce    = 1;
+  uint64 block_height   = 2;
+  string token_contract = 3;
+  string amount         = 4 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
+    (gogoproto.nullable)   = false
+  ];
+  string ethereum_sender = 5;
+  string cosmos_receiver = 6;
+  string orchestrator    = 7;
+}
+
+// WithdrawClaim claims that a batch of withdrawal
+// operations on the bridge contract was executed.
+message WithdrawClaim {
+  uint64 event_nonce    = 1;
+  uint64 block_height   = 2;
+  uint64 batch_nonce    = 3;
+  string token_contract = 4;
+  string orchestrator   = 5;
+}
+
+// This informs the Cosmos module that a logic
+// call has been executed
+message LogicCallExecutedClaim {
+  uint64 event_nonce        = 1;
+  uint64 block_height       = 2;
+  bytes  invalidation_id    = 3;
```

Comment:
why is an ID bytes?

Reply:
right, it should be an integer

Reply:
current gravity design.

Reply:
uint64 ++

Reply:
This is a design decision by @jtremback think of it as a hash value separating different arbitrary logic execution flows.

It can be a number if it really needs to be IMO but Jehan should weigh in.